From 1c5aca5b02b38f4ea04b458c5213f2e19d805b19 Mon Sep 17 00:00:00 2001
From: kbillore <kaushal.billore@intel.com>
Date: Mon, 27 Jul 2020 13:54:08 +0530
Subject: [PATCH] select audio source selection

Recorded video is without audio as Audio source set
is not considering USB as audio input.

Solution: scan and select audio when USB audio input
available.

Change-Id: Ic0af8fb66d88f503f815a7786094a2f04bd8ac9b
Tracked-On: OAM-91515
Signed-off-by: Shiva Kumara shiva.kumara.rudrappa@intel.com
Signed-off-by: kbillore <kaushal.billore@intel.com>
---
 02_0002-Audio-source-selectin.patch     | 56 +++++++++++++++++++++++++
 src/com/android/camera/VideoModule.java | 19 +++++++++
 2 files changed, 75 insertions(+)
 create mode 100644 02_0002-Audio-source-selectin.patch

diff --git a/02_0002-Audio-source-selectin.patch b/02_0002-Audio-source-selectin.patch
new file mode 100644
index 000000000..0ef891042
--- /dev/null
+++ b/02_0002-Audio-source-selectin.patch
@@ -0,0 +1,56 @@
+From 204b86541d824c15968f2af1a23256544c0df29c Mon Sep 17 00:00:00 2001
+From: kbillore <kaushal.billore@intel.com>
+Date: Mon, 27 Jul 2020 13:54:08 +0530
+Subject: [PATCH] Audio source selection when multiple audio source selected
+
+Select right audio source when multiple audio source
+available
+
+Tracked-On : OAM-91515
+Signed-off-by: kbillore <kaushal.billore@intel.com>
+
+Change-Id: Ic0af8fb66d88f503f815a7786094a2f04bd8ac9b
+---
+ src/com/android/camera/VideoModule.java | 19 +++++++++++++++++++
+ 1 file changed, 19 insertions(+)
+
+diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
+index e8222f6d3..a1d4ff62b 100644
+--- a/src/com/android/camera/VideoModule.java
++++ b/src/com/android/camera/VideoModule.java
+@@ -48,6 +48,8 @@ import android.view.KeyEvent;
+ import android.view.View;
+ import android.widget.Toast;
+ 
++import android.media.AudioDeviceInfo;
++
+ import com.android.camera.app.AppController;
+ import com.android.camera.app.CameraAppUI;
+ import com.android.camera.app.LocationManager;
+@@ -1149,6 +1151,23 @@ public class VideoModule extends CameraModule
+ 
+         setRecordLocation();
+ 
++        Context mContext = mActivity.getApplicationContext();
++        AudioManager mAudioManager =
++            (AudioManager)mContext.getSystemService(Context.AUDIO_SERVICE);
++        AudioDeviceInfo[] deviceList =
++            mAudioManager.getDevices(AudioManager.GET_DEVICES_INPUTS);
++        for (AudioDeviceInfo audioDeviceInfo : deviceList) {
++            if (audioDeviceInfo.getType() == AudioDeviceInfo.TYPE_USB_DEVICE) {
++                Log.d(TAG, "Setting preferred device to USB_DEVIVE");
++                mMediaRecorder.setPreferredDevice(audioDeviceInfo);
++                break;
++            } else if (audioDeviceInfo.getType() == AudioDeviceInfo.TYPE_USB_HEADSET) {
++                Log.d(TAG, "Setting preferred device to TYPE_USB_HEADSET");
++                mMediaRecorder.setPreferredDevice(audioDeviceInfo);
++                break;
++            }
++        }
++
+         // Set output file.
+         // Try Uri in the intent first. If it doesn't exist, use our own
+         // instead.
+-- 
+2.17.1
+
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index e8222f6d3..a1d4ff62b 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -48,6 +48,8 @@ import android.view.KeyEvent;
 import android.view.View;
 import android.widget.Toast;
 
+import android.media.AudioDeviceInfo;
+
 import com.android.camera.app.AppController;
 import com.android.camera.app.CameraAppUI;
 import com.android.camera.app.LocationManager;
@@ -1149,6 +1151,23 @@ public class VideoModule extends CameraModule
 
         setRecordLocation();
 
+        Context mContext = mActivity.getApplicationContext();
+        AudioManager mAudioManager =
+            (AudioManager)mContext.getSystemService(Context.AUDIO_SERVICE);
+        AudioDeviceInfo[] deviceList =
+            mAudioManager.getDevices(AudioManager.GET_DEVICES_INPUTS);
+        for (AudioDeviceInfo audioDeviceInfo : deviceList) {
+            if (audioDeviceInfo.getType() == AudioDeviceInfo.TYPE_USB_DEVICE) {
+                Log.d(TAG, "Setting preferred device to USB_DEVIVE");
+                mMediaRecorder.setPreferredDevice(audioDeviceInfo);
+                break;
+            } else if (audioDeviceInfo.getType() == AudioDeviceInfo.TYPE_USB_HEADSET) {
+                Log.d(TAG, "Setting preferred device to TYPE_USB_HEADSET");
+                mMediaRecorder.setPreferredDevice(audioDeviceInfo);
+                break;
+            }
+        }
+
         // Set output file.
         // Try Uri in the intent first. If it doesn't exist, use our own
         // instead.
-- 
2.17.1

